src/Land/Model/soil_water.jl

```diff
@@ -171,29 +214,27 @@ function soil_init_aux!(
 )
     FT = eltype(aux)
     T = get_initial_temperature(land.soil.heat, aux, FT(0.0))
-    S_l = effective_saturation(
-        soil.param_functions.porosity,
-        water.initialϑ_l(aux),
-        soil.param_functions.θ_r,
-    )
-    ψ = pressure_head(
-        water.hydraulics,
-        soil.param_functions,
-        water.initialϑ_l(aux),
-        water.initialθ_i(aux),
-    )
+    θ_r = soil.water.param_functions.θ_r(aux)
+    ν = soil.param_functions.porosity #(aux)
```

Comment:
Is there no need to soil change porosity as a function of soil depth / space?

Reply:
no, youre right that we need that too. We need to change all the soil parameters, but I started with just the ones that are used by the hydrology model. Heat model parameters and porosity (used by both heat and water) will come after. 

on a related note, prior to this PR, all soil parameters were in a single object (`SoilParamFunctions`), but we thought it might be cleaner to split them into heat parameters, water parameters, and joint parameters (which I started to do here). But maybe it is simpler to keep them all in one spot (undo that particular change before merging), and change all params to be functions of aux, all at once, by extending this PR to heat params and porosity too 🤷‍♀️